### ISSUE TITLE
[Feature]: Toggle visibility of items in clipRect

### ISSUE THREAD

**Issue by bitttttten** [fabricjs/fabric.js#10009]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have searched and referenced existing issues, feature requests and discussions
- [X] I am filing a **FEATURE** request.

### Description

We are using clipRect to have a kind of "artboard", where the user can have a full screen canvas.

For example, in the following 3 screenshots I can add some text, drag them to the edge my artboard, and have it cropped.

<img width="916" alt="Screenshot 2024-07-16 at 15 54 59" src="https://github.com/user-attachments/assets/0d0074ec-88fc-4b3f-979c-2d13c4b64fab">
<img width="656" alt="Screenshot 2024-07-16 at 15 55 21" src="https://github.com/user-attachments/assets/d272bbbe-c892-4d77-9f3b-b9a5ab249677">
<img width="653" alt="Screenshot 2024-07-16 at 15 55 24" src="https://github.com/user-attachments/assets/f6b1f8c7-4b6f-4640-beca-9d74542817f0">

This works great for us. However if we drag the text further out, and release, then the item seems to dissappear:

<img width="570" alt="Screenshot 2024-07-16 at 15 55 15" src="https://github.com/user-attachments/assets/6a656ca6-f754-454a-a754-b65003508b03">
<img width="570" alt="Screenshot 2024-07-16 at 15 55 15" src="https://github.com/user-attachments/assets/f5f109a7-a271-41f6-8237-6f8dc1c6533d">

We would love to be able to have some kind of transparency effect or toggle the visibility of the clipPath here.

Does fabric support this.

We are doing:

```ts
    const clipRect = new Rect({
      width: this.width,
      height: this.height,
      absolutePositioned: true,
    })
    const frame = new Group([], {
      layoutManager: new LayoutManager(new ClipPathLayout()),
      selectable: this.selectable,
      subTargetCheck: true,
      interactive: true,
      clipPath: clipRect,
    })
    
    // and now we add items to our frame
```

Any suggestions is appreciated!

### Current State

The current behaviour is that the clipPath is hiding _all_ items, however, for our use case we want to show the items or have them at half transparency.

In other editors, for example, you can have this effect:

<img width="505" alt="Screenshot 2024-07-16 at 15 59 55" src="https://github.com/user-attachments/assets/b8a6ab77-e364-49ac-9e46-bf2d4d38d72f">

**Comment by asturur** [comment]
Instead of a clipPath for the canvas, you can use a custom overlay,
It works way easier and has not the performance degradation of the clipPath.

Maybe i can make this an official example.

**Comment by bitttttten** [comment]
> Maybe i can make this an official example.

That sounds great! If you need any help with it let me know.

It seems setOverlayImage doesn't exist (https://github.com/search?q=repo:fabricjs/fabric.js%20setOverlay&type=code) so I am not sure about which API you are referring to. which API is this? 🤔

**Comment by bitttttten** [comment]
I have tried 

```ts
    const overlayImage = new FabricImage(new window.Image())
    overlayImage.set({ backgroundColor: 'rgba(255,255,255,.25)', width, height })
    overlayImage
    const canvas = new Canvas(this.canvasElem, {
      overlayColor: 'rgba(0,0,0,.25)',
      overlayImage,
      width,
      height,
    })
```

Although this makes all the elements appear faded _in_ the frame. It's a lot closer than the clipRect, which is what I am looking for. 

You can see here for example that the text is faded in the canvas, because the overlayImage is white with 25% opacity.

![Screenshot 2024-07-16 at 21 37 13](https://github.com/user-attachments/assets/d9c32f45-e817-4e11-941f-49e668aa8b09)

**Comment by Clemweb** [comment]
Hi,

@bitttttten just to know how you do with v6 the customs corners button (rounded rectangle on the middles) ?

Can you post an example ?

Thanks

**Comment by ouyoungs** [comment]
> Hi,
> 
> @bitttttten just to know how you do with v6 the customs corners button (rounded rectangle on the middles) ?
> 
> Can you post an example ?
> 
> Thanks

also needed.

**Comment by Clemweb** [comment]
Ok here is my code : 

`Textbox.createControls = () =>
    {
        const controls = controlsUtils.createTextboxDefaultControls();

        // Delete useless handler (for me !)
        delete controls.mt;
        delete controls.mr;

        for (let key in controls)
        {
            controls[key].render = (ctx, left, top, styleOverride, fabricObject) =>
            {
                setResizeHandler(ctx, size, ['ml', 'mr', 'mb', 'mt'].includes(key) ? 'vertical' : 'circle', left, top, fabricObject, key === 'mtr' ? 'rotate' : key);
            }
        }
        
        return {
            controls: {
                ...controls
            }
        }
    }

    FabricObject.createControls = () =>
    {
        const controls = controlsUtils.createObjectDefaultControls();

        for (let key in controls)
        {
            let mode = 'circle';
            switch (key)
            {
                case 'tl':
                case 'tr':
                case 'bl':
                case 'br': mode = 'circle'; break;
                case 'mr':
                case 'ml': mode = 'vertical'; break;
                case 'mt':
                case 'mb': mode = 'horizontal'; break;
            }
            controls[key].render = (ctx, left, top, styleOverride, fabricObject) =>
            {
                setResizeHandler(ctx, size, mode, left, top, fabricObject, key === 'mtr' ? 'rotate' : key);
            }
        }
        
        return {
            controls: {
                ...controls
            }
        }
    }

function setResizeHandler(ctx, size, mode, left, top, fabricObject, type)
{
    ctx.save();
    ctx.translate(left, top);
    ctx.rotate(util.degreesToRadians(fabricObject.angle));

    var x = -size / 2; // Position en x
    var y = -size * 3.5 / 2; // Position en y
    var width = size; // Largeur
    var height = size * 3.5; // Hauteur
    var cornerRadius = size / 2; // Rayon des coins arrondis
    
    if (mode == 'horizontal')
    {
        x = -size * 3.5 / 2; // Position en x
        y = -size / 2; 
        width = size * 3.5;
        height = size;
    }
    else if (mode == 'circle')
    {
        x = 0;
        y = size;
        width = size;
        height = size;
    }

    // Définissez la couleur de remplissage du rectangle
    ctx.fillStyle = "pink";
    ctx.shadowColor = "rgba(0, 0, 0, 0.9)"; // Couleur de l'ombre
    ctx.shadowBlur = 5; // Flou de l'ombre
    
    // Début du chemin
    ctx.beginPath();

    // Middle left, right, top and bottom we draw un flat rounded rectangle
    if (mode !== 'circle')
    {
        
        // Coin supérieur gauche
        ctx.moveTo(x + cornerRadius, y);
        ctx.arcTo(x, y, x, y + cornerRadius, cornerRadius);
        
        // Ligne horizontale supérieure
        ctx.lineTo(x, y + height - cornerRadius);
        
        // Coin inférieur gauche
        ctx.arcTo(x, y + height, x + cornerRadius, y + height, cornerRadius);
        
        // Ligne verticale gauche
        ctx.lineTo(x + width - cornerRadius, y + height);
        
        // Coin inférieur droit
        ctx.arcTo(x + width, y + height, x + width, y + height - cornerRadius, cornerRadius);
        
        // Ligne horizontale inférieure
        ctx.lineTo(x + width, y + cornerRadius);
        
        // Coin supérieur droit
        ctx.arcTo(x + width, y, x + width - cornerRadius, y, cornerRadius);
        
        // Fermer le chemin
        ctx.closePath();

        ctx.fill();
    }
    else
    {
        
        // If rotation handler we draw a svg icon in circle
        if (type === 'rotate')
        {
            ctx.arc(0, 0, size * 1.5, 0, 2 * Math.PI);
            ctx.fill();
            let p = new Path2D("M224 32c123.712 0 224 100.288 224 224h112l-144 144-144-144h112c0-88.365-71.635-160-160-160s-160 71.635-160 160c0 88.366 71.634 160 160 160v64c-123.712 0-224-100.288-224-224s100.288-224 224-224z");
            ctx.scale(size / 250, size / 250);
            ctx.translate(-250, -250);
            ctx.fillStyle = "#585c5f";
            ctx.shadowBlur = 0;
            ctx.fill(p);
            ctx.scale(1, 1);
        }
        else // Juste circle corner handler
        {
            ctx.arc(0, 0, size, 0, 2 * Math.PI);
            ctx.fill();
        }
    }

    // Remplir le rectangle avec des coins arrondis
    
    ctx.restore();
}
`

But there is a little problem with the clickable area on handlers, the custom shpe is not fully clickable look this video : 


https://github.com/user-attachments/assets/24b029c4-a840-4ad4-a584-d55553eb3614

@asturur any idea ?

Thanks

**Comment by Clemweb** [comment]
@asturur Sorry but I didn't find a solution for this last issue. If you know, can you tell me where I need make changes ?
Thanks a lot

**Comment by Clemweb** [comment]
Ok I found the solution : we need to adjust handler size parameter, that it seems to be not linked with handle object. So I increase the handler size and change  my renders of corners to be a little bit smaller.

**Comment by asturur** [comment]
Yes that is correct. i should write and update guides. i don't have time. most of my time for this project is burnt by combing through requests
